[docopt] build error on arm64-osx (shared)

### Operating system

MacOS

### Compiler

Clang 14.0.3

### Steps to reproduce the behavior

```Shell
./vcpkg install docopt
cmake ... DVCPKG_TARGET_TRIPLET=<some dynamic triplet>
```


### Failure logs

```
In file included from <src>/src/main.cpp:8:
In file included from <src>/src/util/cli.h:4:
<src>/build/vcpkg_installed/arm64-osx-dynamic/include/docopt/docopt.h:70:31: error: unknown type name 'dllimport'
        std::map<std::string, value> DOCOPT_API docopt_parse(std::string const& doc,
                                     ^
<src>/build/vcpkg_installed/arm64-osx-dynamic/include/docopt/docopt.h:36:43: note: expanded from macro 'DOCOPT_API'
            #define DOCOPT_API __declspec(dllimport)
                                          ^
<src>/build/vcpkg_installed/arm64-osx-dynamic/include/docopt/docopt.h:70:42: error: expected function body after function declarator
        std::map<std::string, value> DOCOPT_API docopt_parse(std::string const& doc,
                                                ^
<src>/build/vcpkg_installed/arm64-osx-dynamic/include/docopt/docopt.h:83:31: error: unknown type name 'dllimport'
        std::map<std::string, value> DOCOPT_API docopt(std::string const& doc,
                                     ^
<src>/build/vcpkg_installed/arm64-osx-dynamic/include/docopt/docopt.h:36:43: note: expanded from macro 'DOCOPT_API'
            #define DOCOPT_API __declspec(dllimport)
                                          ^
<src>/build/vcpkg_installed/arm64-osx-dynamic/include/docopt/docopt.h:83:42: error: expected function body after function declarator
        std::map<std::string, value> DOCOPT_API docopt(std::string const& doc,
                                                ^
<src>/build/vcpkg_installed/arm64-osx-dynamic/include/docopt/docopt.h:90:15: error: unknown type name 'dllimport'
        std::ostream DOCOPT_API & operator<<(std::ostream&, value const&);
                     ^
<src>/build/vcpkg_installed/arm64-osx-dynamic/include/docopt/docopt.h:36:43: note: expanded from macro 'DOCOPT_API'
            #define DOCOPT_API __declspec(dllimport)
                                          ^
<src>/build/vcpkg_installed/arm64-osx-dynamic/include/docopt/docopt.h:90:28: error: expected function body after function declarator
        std::ostream DOCOPT_API & operator<<(std::ostream&, value const&);
                                  ^
```

### Additional context

Docopt's [source](https://github.com/docopt/docopt.cpp/blob/master/docopt.h) contains the following macro;
```
#ifdef DOCOPT_DLL
        // Whoever is *building* the DLL should define DOCOPT_EXPORTS.
        // The CMakeLists.txt that comes with docopt does this.
        // Clients of docopt.dll should NOT define DOCOPT_EXPORTS.
        #ifdef DOCOPT_EXPORTS
            #define DOCOPT_API __declspec(dllexport)
        #else
            #define DOCOPT_API __declspec(dllimport)
        #endif
    #else
        #define DOCOPT_API
    #endif
#endif
```


AFAIK dllimport / dllexport are only supported by the microsoft compiler / linker, and this will break builds using other compilers.

The [docopt portfile](https://github.com/microsoft/vcpkg/blob/master/ports/docopt/portfile.cmake) replaces `#ifdef DOCOPT_DLL` with `#ifdef 1`, so the MS-specific macro will always be defined, and the port will break dynamic builds on non-windows systems.

As I understand it, dllimport is not strictly required in a shared build -- a quick fix would be to omit it, or have the replacement be contingent on the use of the VS generator. Happy to submit a PR if that sounds good.

--- a/ports/docopt/portfile.cmake
+++ b/ports/docopt/portfile.cmake
@@ -23,7 +23,7 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
     file(READ "${CURRENT_PACKAGES_DIR}/include/docopt/docopt.h" _contents)
-    string(REPLACE "#ifdef DOCOPT_DLL" "#if 1" _contents "${_contents}")
+    string(REPLACE "#ifdef DOCOPT_DLL" "#ifdef _WIN32" _contents "${_contents}")
     file(WRITE "${CURRENT_PACKAGES_DIR}/include/docopt/docopt.h" "${_contents}")
 endif()
 

--- a/ports/docopt/vcpkg.json
+++ b/ports/docopt/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "docopt",
   "version-date": "2022-03-15",
+  "port-version": 1,
   "description": "Command line arguments parser that will make you smile (C++11 port).",
   "license": "MIT OR BSL-1.0",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2302,7 +2302,7 @@
     },
     "docopt": {
       "baseline": "2022-03-15",
-      "port-version": 0
+      "port-version": 1
     },
     "doctest": {
       "baseline": "2.4.11",

--- a/versions/d-/docopt.json
+++ b/versions/d-/docopt.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "9b5aa0bd1e02e5c3a66e56b084c951069a591151",
+      "version-date": "2022-03-15",
+      "port-version": 1
+    },
+    {
       "git-tree": "98cead1c88cad77bef9acf4571d6bd8a2a06cf3a",
       "version-date": "2022-03-15",
       "port-version": 0

